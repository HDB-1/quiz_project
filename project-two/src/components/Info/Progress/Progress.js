import React from 'react'

export default function Progress(props) {
    return (
        <div>
            <h5 id="progress">{props.question.current}/{props.question.total}</h5>
        </div>
    )
}
