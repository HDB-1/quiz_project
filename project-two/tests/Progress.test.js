import React from 'react';
import { shallow} from 'enzyme';
import Progress from '../src/components/Info/Progress/Progress'



describe('Shallow Progress', () => {
  let wrapper;
  
  beforeEach(() => wrapper = shallow(<Progress question={{current:0, total:10}}/>)); // this automatically tests for rendering without crashing.
  it("Should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("Should render one progress tag", () => {
      expect(wrapper.find('#progress').length).toEqual(1);
  });
  it("Should render the correct Progress", () => {
    expect(wrapper.find('#progress').text()).toEqual("1/10");
  });
});
