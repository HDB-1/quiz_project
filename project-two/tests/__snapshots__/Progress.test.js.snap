// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Progress Should render correctly 1`] = `
<div>
  <h5
    id="progress"
  >
    1
    /
    10
  </h5>
</div>
`;
